x-pack/legacy/plugins/siem/server/lib/detection_engine/routes/create_rules_route.ts

```diff
@@ -4,107 +4,158 @@
+        name: schema.string(),
+        reference: schema.string(),
+      }),
+    })
+  ),
+  to: schema.string(),
+  type: schema.oneOf([
+    schema.literal('filter'),
+    schema.literal('query'),
+    schema.literal('saved_query'),
+  ]),
+});
 
-export const createCreateRulesRoute: Hapi.ServerRoute = {
-  method: 'POST',
-  path: DETECTION_ENGINE_RULES_URL,
-  options: {
-    tags: ['access:siem'],
-    validate: {
+export const createRulesRoute = (core: CoreSetup, __legacy: ServerFacade): void => {
+  const router = core.http.createRouter();
+
+  router.post(
+    {
+      path: DETECTION_ENGINE_RULES_URL,
+      validate: { body },
       options: {
-        abortEarly: false,
+        tags: ['access:siem'],
       },
-      payload: createRulesSchema,
     },
-  },
-  async handler(request: RulesRequest, headers) {
-    const {
-      description,
-      enabled,
-      false_positives: falsePositives,
-      filter,
-      from,
-      immutable,
-      query,
-      language,
-      output_index: outputIndex,
-      saved_id: savedId,
-      meta,
-      filters,
-      rule_id: ruleId,
-      index,
-      interval,
-      max_signals: maxSignals,
-      risk_score: riskScore,
-      name,
-      severity,
-      tags,
-      threats,
-      to,
-      type,
-      references,
-    } = request.payload;
-    const alertsClient = isFunction(request.getAlertsClient) ? request.getAlertsClient() : null;
-    const actionsClient = isFunction(request.getActionsClient) ? request.getActionsClient() : null;
-
-    if (!alertsClient || !actionsClient) {
-      return headers.response().code(404);
-    }
-
-    try {
-      if (ruleId != null) {
-        const rule = await readRules({ alertsClient, ruleId });
-        if (rule != null) {
-          return new Boom(`rule_id ${ruleId} already exists`, { statusCode: 409 });
-        }
-      }
-
-      const createdRule = await createRules({
-        alertsClient,
-        actionsClient,
+    async (context, request, response) => {
+      const {
         description,
         enabled,
-        falsePositives,
+        false_positives: falsePositives,
         filter,
         from,
         immutable,
         query,
         language,
-        outputIndex,
-        savedId,
+        output_index: outputIndex,
+        saved_id: savedId,
         meta,
         filters,
-        ruleId: ruleId != null ? ruleId : uuid.v4(),
+        rule_id: ruleId,
         index,
         interval,
-        maxSignals,
-        riskScore,
+        max_signals: maxSignals,
+        risk_score: riskScore,
         name,
         severity,
         tags,
+        threats,
         to,
         type,
-        threats,
         references,
-      });
-      return transformOrError(createdRule);
-    } catch (err) {
-      return transformError(err);
-    }
-  },
-};
+      } = request.body;
+
+      const alertsClient = __legacy.plugins.alerting!.start.getAlertsClientWithRequest(
+        ensureRawRequest(request)
```

Comment:
This is the dead end; we (understandably) can't access the underlying Hapi.Request from a KibanaRequest, which is what we need to retrieve a legacy Alerts client.